tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -1096,6 +1097,11 @@ private[emitter] class FunctionEmitter(jsGen: JSGen) {
 
       require(!allowSideEffects || allowUnpure)
 
+      def testJSArg(tree: TreeOrJSSpread): Boolean = tree match {
+        case JSSpread(_) => false
```

Comment:
Unrelated to this commit, but couldn't we actually recurse here?

Reply:
Yes, I think we could, in ES 2015 mode and with a bit of care. But as you say, this commit preserves the old behavior.

Also that's because we say that `JSSpread` can only be given a JS `Array`, but in ES 2015, `...spread` accepts any iterable (a thing with a `[Symbol.iterator]()` method), so in fact, in ES 2015 mode, `JSSpread` actually accepts any iterable, whose expansion can definitely have side effects.